Mathlib/CategoryTheory/Bicategory/Functor/Cat.lean

```diff
@@ -0,0 +1,179 @@
+
+/-!
+# Pseudofunctors to Cat
+
+In this file, the equalities stated in `CategoryTheory.Bicategory.Functor.Strict`
+for pseudofunctors (from a strict bicategory) are rephrased in the particular
+case the target bicategory is `Cat`. Indeed, in that case, the general lemmas
+for pseudofunctors involve equalities between natural transformations:
+we rephrase them after the application of `NatTrans.app`.
+
+-/
+
+namespace CategoryTheory
+
+open Bicategory
+
+namespace Pseudofunctor
+
+variable {B : Type*} [Bicategory B] (F : Pseudofunctor B Cat)
+
+attribute [local simp] Cat.leftUnitor_hom_app Cat.rightUnitor_hom_app
+  Cat.leftUnitor_inv_app Cat.rightUnitor_inv_app
+  Cat.associator_hom_app Cat.associator_inv_app
+
+section naturality
+
+variable {b₀ b₁ b₂ : B} {X Y : F.obj b₀}
+
+section
+
+variable (f : b₀ ⟶ b₀) (hf : f = 𝟙 b₀) (a : X ⟶ Y)
+
+@[reassoc]
+lemma mapId'_hom_naturality :
+    (F.map f).map a ≫ (F.mapId' f hf).hom.app Y = (F.mapId' f hf).hom.app X ≫ a :=
+  (F.mapId' f hf).hom.naturality a
+
+@[reassoc]
+lemma mapId'_inv_naturality :
+    (F.mapId' f hf).inv.app X ≫ (F.map f).map a = a ≫ (F.mapId' f hf).inv.app Y :=
+  ((F.mapId' f hf).inv.naturality a).symm
+
+end
+
+section
+
+variable (f : b₀ ⟶ b₁) (g : b₁ ⟶ b₂) (fg : b₀ ⟶ b₂)
+  (hfg : f ≫ g = fg) (a : X ⟶ Y)
+
+@[reassoc]
+lemma mapComp'_hom_naturality :
+    (F.map fg).map a ≫ (F.mapComp' f g fg hfg).hom.app Y =
+      (F.mapComp' f g fg hfg).hom.app X ≫ (F.map g).map ((F.map f).map a) :=
+  (F.mapComp' f g fg hfg).hom.naturality a
+
+@[reassoc]
+lemma mapComp'_inv_naturality :
+    (F.map g).map ((F.map f).map a) ≫ (F.mapComp' f g fg hfg).inv.app Y =
+    (F.mapComp' f g fg hfg).inv.app X ≫ (F.map fg).map a :=
+  (F.mapComp' f g fg hfg).inv.naturality a
+
+@[reassoc]
+lemma mapComp'_naturality_1 :
+    (F.mapComp' f g fg hfg).inv.app X ≫
+      (F.map fg).map a ≫ (F.mapComp' f g fg hfg).hom.app Y =
+      (F.map g).map ((F.map f).map a) :=
+  NatIso.naturality_1 (F.mapComp' f g fg hfg) a
+
+@[reassoc]
+lemma mapComp'_naturality_2 :
+    (F.mapComp' f g fg hfg).hom.app X ≫ (F.map g).map ((F.map f).map a) ≫
+      (F.mapComp' f g fg hfg).inv.app Y =
+      (F.map fg).map a :=
+  NatIso.naturality_2 (F.mapComp' f g fg hfg) a
+
+end
+
+end naturality
+
+variable [Strict B]
+
+section unitality
+
+variable {b₀ b₁ : B} (f : b₀ ⟶ b₁) (X : F.obj b₀)
+
+lemma mapComp'_comp_id_hom_app :
+    (F.mapComp' f (𝟙 b₁) f).hom.app X = (F.mapId b₁).inv.app ((F.map f).obj X) := by
+  simp [mapComp'_comp_id]
+
+lemma mapComp'_comp_id_inv_app :
+    (F.mapComp' f (𝟙 b₁) f).inv.app X = (F.mapId b₁).hom.app ((F.map f).obj X) := by
+  simp [mapComp'_comp_id]
+
+lemma mapComp'_id_comp_hom_app :
+    (F.mapComp' (𝟙 b₀) f f).hom.app X = (F.map f).map ((F.mapId b₀).inv.app X) := by
+  simp [mapComp'_id_comp]
+
+lemma mapComp'_id_comp_inv_app :
+    (F.mapComp' (𝟙 b₀) f f).inv.app X = (F.map f).map ((F.mapId b₀).hom.app X) := by
+  simp [mapComp'_id_comp]
```

Comment:
FYI, I have written an attribute to deal with this exact situation! Tagging a lemma, which is an equality between two 2-morphisms, with `@[to_app]` automatically adds a new lemma that specializes the Bicategory where the equality is taking place to `Cat`, applies `NatTrans.app` on both sides and peforms some basic simplification. I wrote this because I was also dealing with Pseudofunctors to `Cat` when making some progress on 2-Yoneda. So I think if you stated these lemmas as `mapComp'_comp_id_hom` in the `Strict` file, you would get these for free by tagging it with `@[to_app]`.

Reply:
So in theory, this file should not need to exist, as you can prove all these lemmas for (strict?) bicategories and tag it with `@[to_app]`. Although the point of making that attribute was to save time, I am not sure if there is any point in making you rewrite these lemmas for strict bicategories just to be able to tag them with `@[to_app]`...

Reply:
So in other words, what I think you should do is add `hom/inv` lemmas to the strict file and tag these with `@[to_app]`.

Reply:
Thanks for the this attribute! However, in other to get suitable lemmas, I had to add a few lemmas #25971 in the simp set to take into account that associators and unitors induce identities.

Reply:
Thanks! Yes I imagined this simp-set was incomplete, and would be added to once people start using the attribute.